### ISSUE TITLE
Django 1.8 changes

### ISSUE THREAD

**Issue by pbanaszkiewicz** [swcarpentry/amy#237]
Hi,

Django 1.8 is out and contains some backward-incompatible changes, for example new stuff for [templates](https://docs.djangoproject.com/en/1.8/ref/templates/upgrading/).

We should check what's working and what has stopped working in Amy with Django 1.8.

We should also decide if we keep Amy compatible with Django 1.7, or Django 1.8, or both.

**Comment by wking** [comment]
On Fri, Apr 03, 2015 at 12:56:44PM -0700, Piotr Banaszkiewicz wrote:
> Django 1.8 is out and contains some backward-incompatible changes…

Previous Django 1.8 discussion in [1,2,3,4].

> We should check what's working and what has stopped working in Amy
> with Django 1.8.

We don't use all that much Django, so we should be able to handle this
by:

* Reading the 1.8 release notes [5] and updating amy accordingly.
* Running the test suite after bumping Django.

> We should also decide if we keep Amy compatible with Django 1.7, or
> Django 1.8, or both.

I'm fine bumping to 1.8 and dropping 1.7 support, but we may want
input from our sysadmin first.  @drio (I think your still the amy
sysadmin ;), do you have any concerns about bumping the stack?  If
you've installed Django using pip I don't foresee any problems
shifting to 1.8.

[1]: https://github.com/swcarpentry/amy/issues/158#issuecomment-71443616
[2]: https://github.com/swcarpentry/amy/pull/221#issuecomment-77890801
[3]: https://github.com/swcarpentry/amy/pull/221#issuecomment-77902272
[4]: https://github.com/swcarpentry/amy/pull/221#issuecomment-77903376
[5]: https://docs.djangoproject.com/en/1.8/releases/1.8/

**Comment by pbanaszkiewicz** [comment]
Issues I found after running the test suite:

1. We need to run migrations (`makemigrations` and then `migrate`) because of https://docs.djangoproject.com/en/1.8/releases/1.8/#abstractuser-last-login-allows-null-values
2. We need to do something about our forms: #240 

After fixing 1. and running the test suite I only had some issues with the forms. I did not check them out so they may or may not be connected to #240.
